src/System.Net.Security/src/System/Net/Security/SecureChannel.cs

```diff
@@ -624,15 +624,22 @@ private bool AcquireClientCredentials(ref byte[] thumbPrint)
         //
         // Acquire Server Side Certificate information and set it on the class.
         //
-        private bool AcquireServerCredentials(ref byte[] thumbPrint)
+        private bool AcquireServerCredentials(ref byte[] thumbPrint, byte[] clientHello)
```

Comment:
it might be possible to use Span<T> here instead of byte[]

Reply:
we would have to convert it everywhere to check if there is any gain from that - I'll leave it out of this PR

Reply:
unless you mean only for our case but still it shouldn't matter